### PR TITLE
Add pocket data to all magazines

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Infrastructure "Add pocket data to all magazines"

#### Purpose of change
Puts data in place to start cutting old code.

#### Describe the solution
Add pocket data to all magazine type items. This data can replace the `type` and `capacity` members of `islot_magazine` going forward.

#### Testing
Game loads, magazines still work.

### CODE REVIEW

File: data/json/items/magazine/liquid.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "ammo_type": [ "flammable" ],
     "capacity": 3000,
     "reliability": 9,
-    "reload_time": 3
+    "reload_time": 3,
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "flammable": 3000 } } ]
```
Comment (KorGgenT) [inline data/json/items/magazine/liquid.json:19]: pretty sure these need to be watertight

File: data/json/items/magazine/weldgas.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
     "capacity": 60,
     "count": 60,
     "reliability": 10,
-    "flags": [ "NO_UNLOAD", "NO_RELOAD" ]
+    "flags": [ "NO_UNLOAD", "NO_RELOAD" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "weldgas": 60 } } ]
```
Comment (KorGgenT) [inline data/json/items/magazine/weldgas.json:20]: i don't know if weldgas is a gas or a liquid, but the pocket needs to be watertight or gastight depending.

**Comment by Inglonias** [comment]
If you're converting magazines to pockets, does that mean you can load things into them that aren't bullets? Do guns check to make sure that the things loaded into magazines are bullets?

**Comment by KorGgenT** [comment]
no, that's what ammo restriction is for

**Pull request merged**